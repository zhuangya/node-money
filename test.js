'use strict';

const m = require('./');

test('valueOf as numebr', () => {
  expect(m(42) + 1).toBe(43);
  expect(m(1) + m(1) + m(1)).toBe(3);
});

test('toString as accounting.js formatted result', () => {
  expect(`price: ${m(42)}`).toBe('price: $42.00');
});

test('should pass options to accounting.js', () => {
  expect(`money in CNY: ${m(42, { symbol: '￥' })}`).toBe('money in CNY: ￥42.00')
  expect(`money in EUR: ${m(4999.99, "€", 2, ".", ",")}`).toBe('money in EUR: €4.999,99')
});
