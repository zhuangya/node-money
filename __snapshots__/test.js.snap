// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valueOf as numebr 1`] = `"43"`;

exports[`valueOf as numebr 2`] = `"3"`;
